src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtils.java

```diff
@@ -173,66 +176,61 @@ public static double parseDouble(String text, int type) {
                     radix = BASE_16;
                     txt = txt.substring(2);
                 }
-                else if (txt.charAt(0) == '0') {
+                else if (txt.startsWith("0b") || txt.startsWith("0B")) {
+                    radix = BASE_2;
+                    txt = txt.substring(2);
+                }
+                else if (CommonUtils.startsWithChar(txt, '0')) {
                     radix = BASE_8;
                     txt = txt.substring(1);
                 }
-                if (CommonUtils.endsWithChar(txt, 'L') || CommonUtils.endsWithChar(txt, 'l')) {
-                    txt = txt.substring(0, txt.length() - 1);
-                }
-                if (!txt.isEmpty()) {
-                    if (type == TokenTypes.NUM_INT) {
-                        result = parseInt(txt, radix);
-                    }
-                    else {
-                        result = parseLong(txt, radix);
-                    }
-                }
+                result = parseNumber(txt, radix, type);
                 break;
             default:
+                result = Double.NaN;
                 break;
```

Comment:
A simple return of zero for unexpected input may lead to a hard to find error.
The NaN is a bit better for this. An exception is even better.